src/RouteAggregator.js

```diff
@@ -0,0 +1,132 @@
+import invariant from 'invariant';
+import Relay from 'react-relay';
+
+import getParamsForRoute from './getParamsForRoute';
+
+export default class RouteAggregator {
+  constructor() {
+    this.uniqueQueryNames = new WeakMap();
```

Comment:
also, how are we messaging to people about that they'll need a WeakMap in older browsers?